tests/manage/rgw/test_obc_quota.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+import pytest
+
+from ocs_ci.ocs.resources.objectbucket import OBC
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.bucket_utils import (
+    list_obc_objects,
+    copy_random_individual_objects,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TestOBCQuota:
+    """
+    Test OBC Quota feature
+    """
+
+    @pytest.mark.parametrize(
+        argnames="amount,interface,quota",
+        argvalues=[
+            pytest.param(
+                *[1, "RGW-OC", {"maxObjects": "1", "maxSize": "2M"}],
+            ),
+        ],
+    )
+    def test_obc_quota(
+        self, awscli_pod_session, rgw_bucket_factory, amount, interface, quota
+    ):
+        """
+        Test OBC quota feature
+            * create OBC with some quota set
+            * check if the quota works
+            * change the quota
+            * check if the new quota works
+        """
+        bucket_name = rgw_bucket_factory(amount, interface, quota=quota)[0].name
+        logger.info("Bucket created: {}".format(bucket_name))
+
+        obc_obj = OBC(bucket_name)
+        full_bucket_path = f"s3://{bucket_name}"
+        amount = int(quota["maxObjects"]) + 1
+        test_dir = "/test_quota/objects_1"
+        out = copy_random_individual_objects(
```

Comment:
yeah there is a reason, If i use `write_random_test_objects_to_bucket` it uses `sync_object_directory` to sync the objects in the directory to the bucket... but what i observed is... when sync function is used... it actually syncs more than the set quota without throwing QuotaExceeded error.. this seems like an issue with s3 sync to me... was about to discuss this with you and verified the issue today again..

> 13:04:16 - MainThread - ocs_ci.ocs.bucket_utils - INFO - Syncing all objects and directories from test_obc_quota[1-RGW-OC-quota0]/result to s3://rgw-oc-bucket-2735aaa8fe534bfab251475479
13:04:16 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage rsh session-awscli-relay-pod-581af30523ed490 sh -c "AWS_CA_BUNDLE=/cert/service-ca.crt AWS_ACCESS_KEY_ID=***** AWS_SECRET_ACCESS_KEY=***** AWS_DEFAULT_REGION=us-east-1 aws s3 --endpoint=***** sync test_obc_quota[1-RGW-OC-quota0]/result s3://rgw-oc-bucket-2735aaa8fe534bfab251475479"
13:04:21 - MainThread - ocs_ci.ocs.bucket_utils - INFO - List objects in s3://rgw-oc-bucket-2735aaa8fe534bfab251475479
13:04:21 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage rsh session-awscli-relay-pod-581af30523ed490 sh -c "AWS_CA_BUNDLE=/cert/service-ca.crt AWS_ACCESS_KEY_ID=***** AWS_SECRET_ACCESS_KEY=***** AWS_DEFAULT_REGION=us-east-1 aws s3 --endpoint=***** ls s3://rgw-oc-bucket-2735aaa8fe534bfab251475479"
13:04:27 - MainThread - test_obc_quota - INFO - List objects:
 2022-03-28 07:34:21    1048576 object-0
2022-03-28 07:34:21    1048576 object-1
2022-03-28 07:34:21    1048576 object-2
2022-03-28 07:34:21    1048576 object-3
2022-03-28 07:34:21    1048576 object-4
2022-03-28 07:34:21    1048576 object-5


Quota set: "maxObjects": "1", "maxSize": "50M"

Reply:
um okay.. Last time I checked, It failed at 20 seconds too.. Ill try to spend some time again and see if it works. i dont think more than 20 seconds wait is ideal thing to do.

Reply:
We could just replace the `sync` with a `cp`, but it seems like you've found a bug, so I think it'd be good to file on in Bugzilla and have the issue fixed rather than worked around

Reply:
Have filed a bug, issue is been moved 4.11.. keeping the cp for now as it works just fine until we have a fix